notebooks/6_search_in_continuous_spaces/index.md

```diff
@@ -0,0 +1,282 @@
+﻿Search in Continuous Space
+#
+# **Table of Contents**
+` `**TOC \o "1-3" \h \z \u [Introduction	 PAGEREF _Toc86421092 \h 2**](#_Toc86421092)**
+
+[**Type of optimization techniques	 PAGEREF _Toc86421093 \h 2**](#_Toc86421093)
+
+[*Linear programming (LP) and Nonlinear programming (NLP)	 PAGEREF _Toc86421094 \h 2*](#_Toc86421094)
+
+[*Continuous optimization and Integer programming (IP)	 PAGEREF _Toc86421095 \h 2*](#_Toc86421095)
+
+[*Constrained optimization and Unconstrained optimization	 PAGEREF _Toc86421096 \h 2*](#_Toc86421096)
+
+[*Differentiable optimization and Non-differentiable optimization	 PAGEREF _Toc86421097 \h 3*](#_Toc86421097)
+
+[Cost functions	 PAGEREF _Toc86421098 \h 3](#_Toc86421098)
+
+[**Convexity	 PAGEREF _Toc86421099 \h 4**](#_Toc86421099)
+
+[*Convex versus non-Convex Optimization Problems	 PAGEREF _Toc86421100 \h 5*](#_Toc86421100)
+
+[*Local and global optimization:	 PAGEREF _Toc86421101 \h 5*](#_Toc86421101)
+
+[Local optimization	 PAGEREF _Toc86421102 \h 5](#_Toc86421102)
+
+[Global optimization	 PAGEREF _Toc86421103 \h 6](#_Toc86421103)
+
+[*Locally optimal for a convex optimization problem	 PAGEREF _Toc86421104 \h 6*](#_Toc86421104)
+
+[*Hessian being positive semi-definite iff convexity (Theorem one: One dimensional)	 PAGEREF _Toc86421105 \h 6*](#_Toc86421105)
+
+[*Hessian being positive semi-definite iff convexity (Theorem two: Multi-dimensional)	 PAGEREF _Toc86421106 \h 7*](#_Toc86421106)
+
+[**Gradient	 PAGEREF _Toc86421107 \h 8**](#_Toc86421107)
+
+[*Gradient Descent	 PAGEREF _Toc86421108 \h 8*](#_Toc86421108)
+
+[*Challenges with gradient descent	 PAGEREF _Toc86421109 \h 10*](#_Toc86421109)
+
+[Local minima and saddle points	 PAGEREF _Toc86421110 \h 10](#_Toc86421110)
+
+[Vanishing and Exploding Gradients	 PAGEREF _Toc86421111 \h 10](#_Toc86421111)
+
+[**Conclusion	 PAGEREF _Toc86421112 \h 11**](#_Toc86421112)
+
+[**References	 PAGEREF _Toc86421113 \h 11**](#_Toc86421113)
+
+
+#
+#
+
+
+# Introduction

```

Comment:
You can provide some examples of optimization problems that can help readers to understand the significance of this topic. (checkout slides)